Automattic/wp-calypso
# Add instructional 'Fonts' section to customizer when fonts are only modifiable from 'Global Styles'

<img src="https://user-images.githubusercontent.com/10274366/124637552-e2970080-de57-11eb-843e-388f10958932.png" width=400px />

---

[What users will see if we can't pick out a page to open the block editor with](https://user-images.githubusercontent.com/10274366/124637710-1540f900-de58-11eb-85a5-0fb0694ed784.png)

### Testing instructions
#### Simple Site Testing Instructions

1. Host a calypso app locally: `cd wp-calypso; yarn && yarn start`
2. Host the editing toolkit plugin as well. `cd wp-calypso/apps/editing-toolkit; yarn dev --sync`
3. Make sure your hosts file is pointing wordpress.com and public-api.wordpress.com to your sandbox, as well as the specific site you're about to test with.
4. Make sure you have an active SSH connection to your sandbox
4. Perform all the following steps using your locally hosted calypso app.
3. Activate a theme which requires fonts to be modified from global styles (such as twenty twenty one)
4. Go to customizer
5. Expand the 'Fonts' section
6. Read the content.
7. Click the link which says 'Click here to open the Block Editor and change your fonts.'
8. It should open your homepage and expand the global styles section
9. This tracks event should have been triggered: calypso_customizer_global_styles_block_editor_link_clicked
10. Click the link which says 'Learn more about changing fonts using Global Styles.'
11. It should open this page: https://wordpress.com/support/custom-fonts/#changing-fonts-with-global-styles
12. This tracks event should have been triggered: calypso_customizer_global_styles_support_link_clicked

#### Atomic Testing Instructions

1. Create an atomic site
2. Make sure your hosts file is pointing to production servers
3. Download the zipped version of the editing toolkit plugin from [here](https://teamcity.a8c.com/viewLog.html?buildId=6417514&buildTypeId=calypso_WPComPlugins_EditorToolKit&tab=artifacts#).
4. Install/activate this plugin (You may need to deactivate the existing editing toolkit plugin first)
3. Activate a theme which requires fonts to be modified from global styles (such as twenty twenty one)
4. Go to customizer
5. Expand the 'Fonts' section
6. Read the content.
7. Click the link which says 'Click here to open the Block Editor and change your fonts.'
8. It should open your homepage in the block editor (The global styles section won't auto expand because we're using the local development environment. But you should have ?openSidebar=globalStyles in the URL you were brought to)
9. This tracks event should have been triggered: calypso_customizer_global_styles_block_editor_link_clicked
10. Click the link which says 'Learn more about changing fonts using Global Styles.'
11. It should open this page: https://wordpress.com/support/custom-fonts/#changing-fonts-with-global-styles
12. This tracks event should have been triggered: calypso_customizer_global_styles_support_link_clicked

#### Test that the new 'fonts' section doesn't show if there is an existing 'fonts' section

1. Host a calypso app locally: `cd wp-calypso; yarn && yarn start`
2. Host the editing toolkit plugin as well. `cd wp-calypso/apps/editing-toolkit; yarn dev --sync`
(NOTE: If you run into any trouble getting toolkit setup, you can refer to [this readme](https://github.com/Automattic/wp-calypso/tree/trunk/apps/editing-toolkit) for an additional information)
3. Make sure your hosts file is pointing wordpress.com to your sandbox, as well as the specific site you're about to test with.
4. Make sure you have an active SSH connection to your sandbox
4. Perform all the following steps using your locally hosted calypso app.
1. Activate a theme which does not support global styles (such as twenty twenty)
2. Go to customizer
3. The 'Fonts' section described above should not exist

https://user-images.githubusercontent.com/10274366/124625215-9cd43b00-de4b-11eb-9c2d-5a79da511bec.mov
